servant/src/Servant/Types/SourceT.hs

```diff
@@ -89,6 +89,12 @@ instance MFunctor SourceT where
     hoist f (SourceT m) = SourceT $ \k -> k $
         Effect $ f $ fmap (hoist f) $ m return
 
+instance (Monad m) => Semigroup (SourceT m a) where
+    (SourceT l) <> (SourceT r) = SourceT $ \k -> l k >> r k
```

Comment:
`>>` ?

Reply:
You're right; fixed in 27e9d33ad14288b8550bc5ee4b00c4f1806e6aa0. Essentially now it composes `k` with appending the right-hand stream.